### PR TITLE
Istio gateway/sidecar proxy image

### PR DESCRIPTION
**Pull request by tcnghia**
This is the dataplane part of Istio, consisting of:
- A custom-built Envoy that contains Istio plugins (Wasm, telemetry)
- iptables to route inbound/outbound traffic through the Envoy proxy when acting as a sidecar
- pilot-agent to bootstrap the Envoy with some Istio-specific configurations

### Image Size

- [x] The Image is smaller in size than its common public counterpart. (163MB v 164MB)
- [ ] The Image is larger in size than its common public counterpart (please explain in the notes).

Notes:

### Image Vulnerabilities

- [ ] The Grype vulnerability scan returned 0 CVE(s).
- [x] The Grype vulnerability scan returned > 0 CVE(s) (please explain in the notes):

Notes:  CVE 2019-3826 is a false positive because of Prometheus versioning of 2.43 = 0.43.

### Basic Testing - K8s cluster
- [x] The container image was successfully loaded into a kind cluster.
- [ ] The container image could not be loaded into a kind cluster (please explain in the notes).

Notes:

### Basic Testing - Package/Application

- [x] The application is accessible to the user/cluster/etc. after start-up.
- [ ] The application is not accessible to the user/cluster/etc. after start-up. (please explain in the notes).

Notes:

### Helm

- [ ] A Helm chart has been provided and the container image can be used with the chart.  If needed, please add a -compat package to close any gaps with the public helm chart.
- [x] A Helm chart has been provided and the container image is not working with the chart (please explain in the notes).
- [ ] A Helm chart was not provided.

Notes: We will need to add the control plane part as well to make this work with the Helm chart.

### Processor Architectures

- [x] The image was built and tested for x86_64.
- [ ] The image could not be built for x86_64 (please explain in the notes).
- [x] The image was built and tested for aarch64.
- [ ] The image could not be built for aarch64. (please explain in the notes).

Notes:

### Functional Testing + Documentation

- [ ] Functional tests have been included and the tests are passing.  All tests have been documnted in the notes section.

Notes:

### Environment Testing + Documentation

- [x] There has not been a request and/or there is no indication that this image needs tested on a public cloud provider.
- [ ] The container image has been tested successfully on a public cloud provider (AWS, GCP, Azure).
- [ ] The container image has not been tested successfully on a public cloud provider (AWS, GCP, Azure) (please explain in the notes).

Notes:

### Version

- [x] The package version is the latest version of the package.  The latest tag points to this version.
- [ ] The package version is the not the latest version of the package (please explain in the notes).

Notes:

### Dev Tag Availability

- [ ] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [ ] There is not a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base') (please explain in the notes).

Notes:

### Access Control + Authentication

- [x] The image runs as `nonroot` and GID/UID are set to 65532 or upstream default
- [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
- [ ] The image requires a non-standard username or non-standard GID/UID (please explain in the notes).

### ENTRYPOINT

- [x] applications/servers/utilities set to call main program with no arguments e.g. [redis-server]
- [ ] applications/servers/utilities not set to call main program with no arguments e.g. [redis-server] (please explain in the notes)
- [ ] base images leave empty.
- [ ] base image and not empty (please explain in the notes).
- [ ] dev variants is set to entrypoint script that falls back to system.
- [ ] dev variants is not set to entrypoint script that falls back to system (please explain in the notes).

### CMD

- [x] For server applications give arguments to start in daemon mode (may be empty)
- [ ] For utilities/tooling bring up help e.g. `–help`
- [ ] For base images with a shell, call it e.g. [/bin/sh]

### Environment Variables

- [x] Environment variables added.
- [ ] Environment variables not added and not required.

### SIGTERM

- [x] The image responds to SIGTERM (e.g., `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`)

### Logs

- [x] Error logs write to stderr and normal logs to stdout. Logs DO NOT write to file.

### Documentation - README

- [x] A README file has been provided and it follows the README template.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -361,6 +361,12 @@ module "influxdb" {
   target_repository = "${var.target_repository}/influxdb"
 }
 
+module "istio-proxy" {
+  source            = "./images/istio-proxy"
+  target_repository = "${var.target_repository}/istio-proxy"
+}
```
Comment (mattmoor) [inline main.tf:367]: Can we create a single Istio directory and have this be a proxy component within it?

I assume we'll have multiple containers at some point.

File: images/istio-proxy/main.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+module "tagger" {
+  source = "../../tflib/tagger"
+
+  depends_on = [module.test-latest]
+
+  tags = merge(
+    { for t in toset(concat(["latest"], module.version-tags.tag_list)) : t => module.latest.image_ref },
+  )
+}
```
Comment (mattmoor) [inline images/istio-proxy/main.tf:39]: Can we start this without version tags (in the new style)?

File: main.tf (hunk 1 above)
Comment (mattmoor) [reply main.tf:367]: e.g. I could imagine us adding an `istioctl` image at least, but `istiod` seems like another logical one.

File: main.tf [hunk 3]
```diff
@@ -361,6 +361,12 @@ module "influxdb" {
...
+
+
```
Comment (mattmoor) [inline main.tf:369]: nit: extra space

File: images/istio-proxy/main.tf (hunk 2 above)
Comment (tcnghia) [reply images/istio-proxy/main.tf]: Done

File: main.tf (hunk 1 above)
Comment (tcnghia) [reply main.tf]: Done

File: main.tf (hunk 3 above)
Comment (tcnghia) [reply main.tf:369]: Done

File: images/istio/configs/proxy/template.apko.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- path: /run
+  type: directory
+  permissions: 0o755
+
+accounts:
+  groups:
+    - groupname: nonroot
+      gid: 65532
+  users:
+    - username: nobody
```
Comment (mattmoor) [inline images/istio/configs/proxy/template.apko.yaml:20]: duplicate username

File: images/istio/image.yaml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+versions:
+  - apko:
+      config: proxy/configs/latest.apko.yaml
```
Comment (mattmoor) [inline images/istio/image.yaml:3]: nit: configs/proxy/latest.apko.yaml, but this isn't semantic anymore :)

File: main.tf (hunk 3 above)
Comment (mattmoor) [reply main.tf:369]: lol it is still there

File: images/istio/configs/proxy/template.apko.yaml (hunk 4 above)
Comment (tcnghia) [reply images/istio/configs/proxy/template.apko.yaml:20]: Done

File: images/istio/image.yaml (hunk 5 above)
Comment (tcnghia) [reply images/istio/image.yaml]: Done

**Comment by tcnghia** [comment]
This is RFAL!

**Review by mattmoor — APPROVED** [review]

**Pull request merged**